tests/unit/s2n_ktls_test.c

```diff
@@ -13,13 +13,61 @@
  * permissions and limitations under the License.
  */
 
+#include "tls/s2n_ktls.h"
+
 #include "crypto/s2n_cipher.h"
+#include "error/s2n_errno.h"
+#include "s2n.h"
 #include "s2n_test.h"
+#include "stdio.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_handshake_type.h"
+#include "utils/s2n_safety.h"
+
+#define S2N_TEST_INADDR_LOOPBACK 0x7f000001 /* 127.0.0.1 */
+
+S2N_RESULT s2n_ktls_validate(struct s2n_connection *conn);
+S2N_RESULT s2n_ktls_validate_socket_mode(struct s2n_connection *conn, s2n_ktls_mode ktls_mode);
+S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
+S2N_RESULT s2n_ktls_configure_socket(struct s2n_connection *conn, s2n_ktls_mode ktls_mode);
+S2N_RESULT s2n_ignore_ktls_ulp_for_testing(void);
+
+S2N_RESULT s2n_test_configure_ktls_connection(struct s2n_connection *conn, int *fd)
+{
+    *fd = 1;
+    EXPECT_SUCCESS(s2n_connection_set_write_fd(conn, *fd));
+    EXPECT_SUCCESS(s2n_connection_set_read_fd(conn, *fd));
+
+    conn->managed_send_io = true;
+    conn->ktls_send_enabled = false;
+    conn->managed_recv_io = true;
+    conn->ktls_recv_enabled = false;
+
+    conn->initial->cipher_suite->record_alg = &s2n_record_alg_aes128_gcm;
+    conn->actual_protocol_version = S2N_TLS12;
+
+    return S2N_RESULT_OK;
+}
 
 int main(int argc, char **argv)
 {
     BEGIN_TEST();
 
+    EXPECT_OK(s2n_ignore_ktls_ulp_for_testing());
+
+    /* kTLS feature probe */
+    {
+#if defined(__linux__)
+    #include "linux/version.h"
+    /* kTLS support was first added to AL2 starting in 5.10.130. */
+    #if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 130))
```

Comment:
And our CI includes AL2 > 5.10.130? This check gets exercised? You've confirmed the CI fails if you check for S2N_PLATFORM_SUPPORTS_KTLS being set instead?

Reply:
I am checking this now to see if we are able to test ktls platform supported in our CI at the moment. We will need this eventually to deliver kTLS but it doent need to happen in this PR. Will report back

Reply:
```
s2nValgrindOpenSSL111Gcc9
kTLS SUPPORTED!!!! (s2n_ktls_feature_probe_test.c:24)
```
- CI Image: 024603541914.dkr.ecr.us-west-2.amazonaws.com/docker:ubuntu18codebuild
- [CI run](https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/s2nGeneralBatch/build/s2nGeneralBatch%3A7df748a6-057d-47f3-bd43-382517998c84/?region=us-west-2)
- [branch](https://github.com/toidiu/s2n-tls/blob/ak-test/tests/unit/s2n_ktls_feature_probe_test.c)

Reply:
To be clear here, its possible to check this but we **dont currently** since out AL2 image doesnt support kTLS. We dont because of the `#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 130))` check.

Solution:
- detect and ignore AL2
- update AL2 image to >5.10.130

Adding a task to the tracking issue: https://github.com/aws/s2n-tls/issues/3711